pandas/tests/frame/indexing/test_insert.py

```diff
@@ -104,3 +106,28 @@ def test_insert_frame(self):
         msg = r"Expected a 1D array, got an array with shape \(2, 2\)"
         with pytest.raises(ValueError, match=msg):
             df.insert(1, "newcol", df)
+
+    @pytest.mark.parametrize("allow_duplicates", [True, False])
+    def test_insert_warning_timezone_naive_and_aware_index(self, allow_duplicates):
+        # GH#44582
+        df = DataFrame(
+            [[1, 2], [3, 4]],
+            columns=date_range("1/1/2013", "1/2/2013"),
+        )
+
+        with tm.assert_produces_warning(FutureWarning):
+            df.insert(
+                loc=0,
+                column=Timestamp("2012-12-30", tz="UTC"),
```

Comment:
i thought the case you were interested in was the reverse: `column` being tznaive and `df.columns` being tzaware.  am i misunderstanding something?

Reply:
Does not really matter, both cases are showing the Warning, wenn allow duplicates is set to false

Reply:
OK thanks